src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/MSBuildShellOutNuGetProject.cs

```diff
@@ -326,6 +355,51 @@ public static string GetMSBuildPath(IVsBuildPropertyStorage buildPropertyStorage
             return null;
         }
 
+        public bool IsRestoreRequired(
+            IEnumerable<VersionFolderPathResolver> pathResolvers,
+            ISet<PackageIdentity> packagesChecked,
+            ExternalProjectReferenceContext context)
+        {
+            return true;
```

Comment:
Yes, good idea! I think whoever implements the real CPS project type will use this as some reference. I reminder would be good.